main.go

```diff
@@ -158,38 +170,220 @@ func main() {
 		),
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(
-			err, "unable to create controller", "controller",
+			err, "Unable to create controller", "controller",
 			"AerospikeCluster",
 		)
 		os.Exit(1)
 	}
 
 	if err = (&asdbv1.AerospikeCluster{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "v1-webhook", "AerospikeCluster")
+		setupLog.Error(err, "Unable to create webhook", "v1-webhook", "AerospikeCluster")
 		os.Exit(1)
 	}
 	// +kubebuilder:scaffold:builder
 
 	if err := mgr.AddHealthzCheck("health", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up health check")
+		setupLog.Error(err, "Unable to set up health check")
 		os.Exit(1)
 	}
 
 	if err := mgr.AddReadyzCheck("check", healthz.Ping); err != nil {
-		setupLog.Error(err, "unable to set up ready check")
+		setupLog.Error(err, "Unable to set up ready check")
 		os.Exit(1)
 	}
 
+	go migrateAerospikeClusters(client, &options, watchNs)
+
 	setupLog.Info("Starting manager")
 
 	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		setupLog.Error(err, "problem running manager")
+		setupLog.Error(err, "Problem running manager")
 		os.Exit(1)
 	}
 
 	eventBroadcaster.Shutdown()
 }
 
+func migrateAerospikeClusters(client crClient.Client, options *ctrl.Options, watchNs string) {
+	for {
+		// Checking if controller is ready.
+		readyzURL := fmt.Sprintf("http://localhost%s/readyz", options.HealthProbeBindAddress)
+
+		resp, err := http.Get(readyzURL) //nolint:gosec // locally created url
+		if err != nil {
+			setupLog.Error(err, "Unable to do ready check")
+			os.Exit(1)
+		}
+
+		if resp.StatusCode == http.StatusOK {
+			setupLog.Info("Controller is ready")
+			resp.Body.Close()
+
+			break
+		}
+
+		setupLog.Info("Controller is not ready, retrying...")
+		resp.Body.Close()
```

Comment:
How many retry it usually takes?...should we add a small sleep before retrying?

Reply:
Actually, I have not seen a single retry in my many runs. This is mostly a safe check.